### ISSUE TITLE
Cloak seems detected by Iran Gov firewall

### ISSUE THREAD

**Issue by Evolve6996** [net4people/bbs#327]
hello, recently (like a week or ago) there was a mass blockage of VPNs in my country, I am using openvpn+Cloak, a few days ago my domain address was blocked by the firewall, so I started investigating how they detected it.

last night, first my connection throttled and after a few hours, my VPS IP address was blocked!

I cannot say it 100% because of the cloak but it looks like the cloak is detected, cause symptoms happened when I was mostly using my cloak+openvpn server.

**is Cloak using version 112 of Firefox maybe they just somehow figured this out because of the old fingerprints of the cloak?**

**Comment by irgfw** [comment]
Yes. SS+Cloak is detected and blocked on most of Iran's ISPs. (MCI/MTN/TCI).
But with a super clean IP address and limited users/traffic it would be ok as well.
The main problem with Cloak is the browser fingerprinting is not updated. with a small update on them, "maybe" it could work again.

**Comment by RPRX** [comment]
Cloak 的 Client Hello 存在设计失误，可以被检测出异常，与具体的指纹无关：https://github.com/net4people/bbs/issues/287#issuecomment-1718887813

但是我们收到了 REALITY 被伊朗 GFW 封锁的报告 https://github.com/XTLS/Xray-core/issues/2778 ，并且 Cloak 似乎没有过于流行以至于会被针对，所以 Cloak 的上述特征可能未被伊朗 GFW 利用，更有可能是伊朗 GFW 在针对 IP、域名、流量特征等。

此外，我们收到的一份秘密报告称，若对 REALITY 进行特殊的设置，伊朗 GFW 就不会封锁它，但报告者不希望公开这个方法。

**Comment by irgfw** [comment]
We are testing Reality heavily in Iran. and we know what that special settings are. but after testing them it showed that the IP history and the cleanliness of it is more important that some security `(firewall/iptables/ufw/...)` settings.

**Comment by irgfw** [comment]
Because the trigger is not based on Usage. it's based on packet-size and some traffic characteristics.

**Comment by markpash** [comment]
@irgfw @Evolve6996 You should consider it's both. There could be any number of heuristics used to determine if an IP should be blocked. We don't know, and while there has been some scientific testing, because we don't control all the variables, it's hard to say otherwise. There's so many different ways someone could think of to try and trigger a block. So let's not discount anything so easily.
